packages/twenty-front/src/modules/views/view-picker/components/ViewPickerCreateOrEditContent.tsx

```diff
@@ -104,6 +108,7 @@ export const ViewPickerCreateOrEditContent = () => {
   );
 
   const onIconChange = ({ iconKey }: { iconKey: string }) => {
+    toggleTakeControlOfCustomIcon(true);
```

Comment:
instead of introducing this "takeControlOfCustomIcon" logic, why not setting the viewPickerSelectedIcon on dropdown opening.
I would do that in ViewPickerCreateOrEditContentEffect, we already set the Icon from there for editView case, we can set it to table or kanban in case of createView